### ISSUE TITLE
[a11y]: StructuredListRow does not have a 'row' role

### ISSUE THREAD

**Issue by wolf-adam** [carbon-design-system/carbon#10497]
### Package

carbon-components-react

### Browser

Chrome

### Operating System

MacOS

### Package version

7.50.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker Version 3.1.14

### Description

In PAL's **ObservePlanTile** the **EditPlanSidePanel** children uses a **StructuredListRow** component, which has several other **StructuredListCell** children.
At the moment the **StructuredListRow** has the `label` property and the rows can get selected. Because it is a label instead of the div, the **Accessibility Checker** alerts us about the missing role.

If we delete the `label` property, the row will become unselectable, but the violation would be solved.

Image from the original issue itself:
![Screenshot 2022-01-21 at 16 16 22](https://user-images.githubusercontent.com/38081038/150551809-7d4f2564-d6a0-472f-89dc-d79ff96c5a6e.png)

### WCAG 2.1 Violation

IBM 4.1.2 Name, Role, Value

### CodeSandbox example

https://codesandbox.io/s/structuredlistrow-a11y-nq66f-nq66f?file=/src/index.js

### Steps to reproduce

It should be visible inside the codeSandbox example right away, but it looks like it can not check this issue inside the iframe tag

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by laurenmrice** [comment]
Hey @wolf-adam , this issue will be fixed in v11 but we are unsure if this is something we can fix in v10 because it is a breaking change. We will look into it. 👍

**Comment by wolf-adam** [comment]
Thanks @laurenmrice!
I'll update our issue with this information.

**Comment by mashenka123** [comment]
@laurenmrice Could you please kindly confirm when this could get a fix? Thank you!

**Comment by laurenmrice** [comment]
Update: This will get fixed as a part of our v11 release, which we will be shipping at the end of March.

**Comment by mashenka123** [comment]
@laurenmrice Shall we see the fix this week? Thank you!

**Comment by mashenka123** [comment]
@laurenmrice Could you please kindly confirm if this is fixed?

**Comment by tay1orjones** [comment]
@mashenka123 The violation is not present in the v11/current storybook, and the rows now have a `role="row"`

![image](https://user-images.githubusercontent.com/3360588/162029709-fdfd631a-6ba6-476a-9110-d4ee45f23b7c.png)

The 2 violations listed there are false positives from the storybook environment and not the component itself.

**Comment by xiepingp** [comment]
Thanks for the fixing! The violation related to the 'row' role no longer exists:
![Screen Shot 2022-04-07 at 12 39 40 PM](https://user-images.githubusercontent.com/63960906/162121507-9c755854-8af5-4b61-a60c-70840b99c72d.png)
